core/lib/zksync_core_leftovers/src/lib.rs

```diff
@@ -741,6 +748,37 @@ pub async fn initialize_components(
         .context("add_tee_verifier_input_producer_to_task_futures()")?;
     }
 
+    if eth
+        .sender
+        .clone()
+        .context("eth_sender")?
+        .pubdata_sending_mode
+        == PubdataSendingMode::Custom
+    {
+        if !components.contains(&Component::DADispatcher) {
+            panic!("DA dispatcher requires custom pubdata sending mode");
+        }
```

Comment:
- Please use `anyhow` tooling to error in this case rather than panicking (e.g., using `anyhow::ensure!`).
- Isn't the check the other way around, i.e., custom pubdata sending mode requiring the DA dispatcher?

Reply:
1. done
2. yeah, reversed it